app/models/request.rb

```diff
@@ -9,18 +9,28 @@ class Request < ApplicationRecord
   validates :day_begin, presence: true
   validates :day_end, presence: true
 
-  validate :begin_end_check, if: ->{day_begin || day_end}
-  validate :begin_check, if: :day_begin
+  validate :begin_end_check, if: ->{day_begin || day_end}, on: :create
+  validate :begin_check, if: :day_begin, on: :create
 
-  scope :order_day_begin, ->{order day_begin: :asc}
+  scope :order_day_begin, ->{order :day_begin}
+
+  after_save :update_quantity, if: :fulfilled?
+
+  UPDATE_ATTRS = %i(status note).freeze
 
   private
 
   def begin_end_check
-    errors.add(:day_end, t(".error_begin_end")) if day_begin > day_end
+    errors.add(:day_end, I18n.t("error_begin_end")) if day_begin > day_end
   end
 
   def begin_check
-    errors.add(:day_begin, t(".error_begin")) if day_begin < Time.zone.now
+    errors.add(:day_begin, I18n.t("error_begin")) if day_begin < Time.zone.now
+  end
+
+  def update_quantity
+    books.each do |book|
+      book.update quantity: book.quantity - 1
```

Comment:
chỗ này nếu em dùng callback là before_update và dùng dạng book.quantity = book.quantity - 1

sau đó nó sẽ dc update vào DB khi em save request  [ở đây](https://github.com/awesome-academy/hcm-fresher-libraries_management_system/pull/17/files#diff-8eb13db90b3be1246b13bd59f2b0fd11e3171296c1e2b34aec92f56cd0627ea5R13) thì ổn 

con nếu như em hiện tại thì vì có 2 thao tác update DB nên em dùng transaction - bọc lạis thành công thì thành công hết ất ... 
và dùng update! cơ

Reply:
Dạ e đọc một hồi mới hiểu @@, ko biết e fix đúng ý chị nói hk